vocode/streaming/transcriber/deepgram_transcriber.py

```diff
@@ -24,48 +25,66 @@
 )
 from vocode.streaming.utils.worker import AsyncWorker
 from websockets.client import WebSocketClientProtocol
+from numpy.typing import DTypeLike
+from pydantic import BaseModel
 
 PUNCTUATION_TERMINATORS = [".", "!", "?"]
 MAX_SILENCE_DURATION = 2.0
 NUM_RESTARTS = 5
 
 # Silero VAD setup
+pp = PrettyPrinter(indent=2, width=80)
 model, utils = torch.hub.load(repo_or_dir="snakers4/silero-vad", model="silero_vad")
 (get_speech_timestamps, save_audio, read_audio, VADIterator, collect_chunks) = utils
 
-# Constants for Silero VAD
-USE_INT16 = False
-SAMPLE_RATE = 16000 if USE_INT16 else 8000
-DTYPE = np.int16 if USE_INT16 else np.int8
-CHUNK = 512 if USE_INT16 else 256
-WINDOWS = 3
-WINDOW_SIZE = WINDOWS * SAMPLE_RATE // CHUNK
-PREFIX_PADDING_MS = 150  # Minimum speech duration to trigger detection
-GRACE_PERIOD_MS = 300  # Grace period before resetting speech detection
+
+class AudioEncodingModel(BaseModel):
```

Comment:
![image](https://github.com/user-attachments/assets/7d581dda-6176-477a-a0f4-1f5ffb52fac5)
Will fix it so it references the source of truth, didn't want to change the structure too much from the original code because I already changed a bunch in this pr. Also not sure the vad was working perfectly, but didn't understand it fully so wanted to make it easy to debug and fix it.

Reply:
Yeah you are right, I'll move it to the vad class, and have that as the source of truth, and do downsampling in the vad receive audio, and is_silence calc in vad receive audio